rotkehlchen/api/rest.py

```diff
@@ -4125,3 +4120,115 @@ def get_all_counterparties(self) -> Response:
                 'message': '',
             },
         )
+
+    def get_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            addresses: Optional[List[ChecksumEthAddress]],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        entries = db_addressbook.get_addressbook_entries(
+            book_type=book_type,
+            addresses=addresses,
+        )
+        serialized = [entry.serialize() for entry in entries]
+        return api_response(_wrap_in_ok_result(serialized))
+
+    def add_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            entries: List[AddressbookEntry],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.add_addressbook_entries(book_type=book_type, entries=entries)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+    def update_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            entries: List[AddressbookEntry],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.update_addressbook_entries(book_type=book_type, entries=entries)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+    def delete_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            addresses: List[ChecksumEthAddress],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.delete_addressbook_entries(book_type=book_type, addresses=addresses)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+    def search_for_names_everywhere(self, addresses: List[ChecksumEthAddress]) -> Response:
+        """This method searches for all names of provided addresses known to rotki. We can show
+        only one name per address, and thus we prioritize known names. Priority is the following:
```

Comment:
Make the priority part of the code, so that it eventually becomes customizable.

Reply:
I think I don't get you. What do you mean by making them part of the code?